Tests/BraveWalletTests/AssetDetailStoreTests.swift

```diff
@@ -0,0 +1,510 @@
+// Copyright 2023 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+import Combine
+import XCTest
+import BraveCore
+@testable import BraveWallet
+
+class AssetDetailStoreTests: XCTestCase {
+  private var cancellables: Set<AnyCancellable> = .init()
+  
+  func testUpdateWithBlockchainToken() {
+    let currencyFormatter = NumberFormatter().then { $0.numberStyle = .currency }
```

Comment:
nit: It might be easier to separate the services initialization into a function like we do in a couple other tests, passing in parameters. Helps avoid duplicate lines and all of the `._addObserver = { _ in }` for every observed service.

Future planning we could look to add `BraveWalletTests` target specific helpers for this that every `XCTestCase` can use, instead of a `setupServices()` function in every test class 🫠.